pipeline/03-evaluate.R

```diff
@@ -142,75 +139,69 @@ gen_agg_stats <- function(data, truth, estimate, bldg_sqft,
-        .names = "prior_far_yoy_pct_chg_{.fn}"
-      ),
+      sum_rsf_lst = sum_fns_list %>%
+        set_names(paste0("prior_far_fmv_", names(.))) %>%
+        map(., \(f) exec(f, {{ rsf_col }})) %>%
+        list(),
+      sum_rsf_sf_lst = sum_sqft_fns_list %>%
+        set_names(paste0("prior_far_fmv_per_sqft_", names(.))) %>%
+        map(., \(f) exec(f, {{ rsf_col }}, {{ bldg_sqft }})) %>%
+        list(),
+      yoy_rsf_lst = yoy_fns_list %>%
+        set_names(paste0("prior_far_yoy_pct_chg_", names(.))) %>%
+        map(., \(f) exec(f, {{ estimate }}, {{ rsf_col }})) %>%
+        list(),
       prior_near_num_missing = sum(is.na({{ rsn_col }})),
-      across(
-        .fns = sum_fns_list, {{ rsn_col }},
-        .names = "prior_near_fmv_{.fn}"
-      ),
-      across(
-        .fns = sum_sqft_fns_list, {{ rsn_col }}, {{ bldg_sqft }},
-        .names = "prior_near_fmv_per_sqft_{.fn}"
-      ),
-      across(
-        .fns = yoy_fns_list, {{ estimate }}, {{ rsn_col }},
-        .names = "prior_near_yoy_pct_chg_{.fn}"
-      ),
+      sum_rsn_lst = sum_fns_list %>%
+        set_names(paste0("prior_near_fmv_", names(.))) %>%
+        map(., \(f) exec(f, {{ rsn_col }})) %>%
+        list(),
+      sum_rsn_sf_lst = sum_sqft_fns_list %>%
+        set_names(paste0("prior_near_fmv_per_sqft_", names(.))) %>%
+        map(., \(f) exec(f, {{ rsn_col }}, {{ bldg_sqft }})) %>%
+        list(),
+      yoy_rsn_lst = yoy_fns_list %>%
+        set_names(paste0("prior_near_yoy_pct_chg_", names(.))) %>%
+        map(., \(f) exec(f, {{ estimate }}, {{ rsn_col }})) %>%
+        list(),
 
       # Summary stats of estimate value and estimate per sqft
       estimate_num_missing = sum(is.na({{ estimate }})),
-      across(
-        .fns = sum_fns_list, {{ estimate }},
-        .names = "estimate_fmv_{.fn}"
-      ),
-      across(
-        .fns = sum_sqft_fns_list, {{ estimate }}, {{ bldg_sqft }},
-        .names = "estimate_fmv_per_sqft_{.fn}"
-      ),
+      sum_est_lst = sum_fns_list %>%
+        set_names(paste0("estimate_fmv_", names(.))) %>%
+        map(., \(f) exec(f, {{ estimate }})) %>%
+        list(),
+      sum_est_sf_lst = sum_sqft_fns_list %>%
+        set_names(paste0("estimate_fmv_per_sqft_", names(.))) %>%
+        map(., \(f) exec(f, {{ estimate }}, {{ bldg_sqft }})) %>%
+        list(),
       .groups = "drop"
     ) %>%
     ungroup() %>%
-    # COD, PRD, and PRB all output to a list. We can unnest each list to get
-    # additional info for each stat (95% CI, sample count, etc)
-    tidyr::unnest_wider(cod) %>%
-    tidyr::unnest_wider(COD_CI, names_sep = "_") %>%
-    tidyr::unnest_wider(prd) %>%
-    tidyr::unnest_wider(PRD_CI, names_sep = "_") %>%
-    tidyr::unnest_wider(prb) %>%
-    tidyr::unnest_wider(PRB_CI, names_sep = "_") %>%
-    # Rename columns resulting from unnesting
-    rename_with(~ gsub("%", "", gsub("\\.", "_", tolower(.x))))
+    unnest_wider(ends_with("_lst"))
```

Comment:
This unnesting code _is_ needed as it expands the list columns made by each of the function map calls above.